.github/workflows/container-build-push.yaml

```
@@ -90,11 +93,24 @@ jobs:
           restore-keys: |
             rust-buildkit-cache-${{ matrix.platform }}-
 
+      - name: Read Rust channel
+        id: rust-channel
+        run: |
+          python3 <<'EOF' >"${GITHUB_ENV}"
+          from tomllib import loads
+          from pathlib import Path
+
+          rust_toolchain = loads(Path("./rust-toolchain.toml").read_text())
+          print(f"RUST_VERSION={rust_toolchain['toolchain']['channel']}")
```

nit:
https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md
```suggestion
          print(f"RUST_VERSION={rust_toolchain["toolchain"]["channel"]}")
```